### PR TITLE
[1LP][RFR] Add test to verify service Chargeback in SUI

### PR DESCRIPTION
**Pull request by nachandr**
{{pytest: -v --long-running cfme/tests/ssui/test_ssui_dashboard.py -k 'test_monthly_charges' }}

PRT Results:
Test passed on 58

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit e4b36c2cb1318d7c85741139d02fdf552ee8da9b:

`cfme/tests/ssui/test_ssui_dashboard.py`:
- :bangbang: Line 111:5: **E115** *expected an indented block (comment)*
- :bangbang: Line 125:50: **E999** *SyntaxError: invalid syntax*
- :bangbang: Line 126:18: **E111** *indentation is not a multiple of four*
- :bangbang: Line 126:18: **E113** *unexpected indentation*
- :bangbang: Line 127:18: **E111** *indentation is not a multiple of four*
- :bangbang: Line 128:18: **E111** *indentation is not a multiple of four*
- :bangbang: Line 129:18: **E111** *indentation is not a multiple of four*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/ssui/test_ssui_dashboard.py [hunk 1]
```diff
@@ -27,6 +32,113 @@
...
+        yield desc
+    finally:
+        compute.delete()
+        storage.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def assign_chargeback_rate(provider, new_compute_rate):
+    # Assign custom Compute rate to the Enterprise and then queue the Chargeback report.
+    description = new_compute_rate
```
[the remark this answers is not in the document]
Comment (nachandr) [reply cfme/tests/ssui/test_ssui_dashboard.py]: @rlbabyuk , It's necessary that some some Chargeback rate be assigned  in order to generate Chargeback reports.

[the remark this answers is not in the document]
Comment (nachandr) [reply cfme/tests/ssui/test_ssui_dashboard.py]: @rlbabyuk , I've addressed all the comments.

File: cfme/tests/ssui/test_ssui_dashboard.py [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
 from cfme.services.dashboard import Dashboard
 from cfme import test_requirements
 
+from cfme.utils.log import logger
 from cfme.utils import testgen
 from cfme.utils.version import current_version
-from cfme.utils.appliance import get_or_create_current_appliance, ViaSSUI
+from cfme.utils.appliance import ViaSSUI
+from cfme.utils.wait import wait_for
+from datetime import date
```
Comment (quarckster) [inline cfme/tests/ssui/test_ssui_dashboard.py]: **REQUIRED**
```python
import fauxfactory
import pytest
from datetime import date

from cfme import test_requirements
from cfme.infrastructure.provider import InfraProvider
from cfme.intelligence.chargeback import assignments as cb, rates
from cfme.services.dashboard import Dashboard
from cfme.utils import testgen
from cfme.utils.appliance import ViaSSUI
from cfme.utils.log import logger
from cfme.utils.version import current_version
from cfme.utils.wait import wait_for
```

File: cfme/tests/ssui/test_ssui_dashboard.py [hunk 3]
```diff
@@ -27,6 +32,108 @@
...
+        compute.create()
+        storage = rates.StorageRate(description=desc, fields={
+            'Used Disk Storage': {'per_time': 'Hourly', 'variable_rate': '3'},
+            'Allocated Disk Storage': {'per_time': 'Hourly', 'fixed_rate': '3'}})
+        storage.create()
+        yield desc
+
+    finally:
+        compute.delete()
+        storage.delete()
```
Comment (quarckster) [inline cfme/tests/ssui/test_ssui_dashboard.py]: Please explain this `try finally` block.

File: cfme/tests/ssui/test_ssui_dashboard.py [hunk 4]
```diff
@@ -27,6 +32,108 @@
...
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': description}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+    logger.info('Assigning CUSTOM Compute and Storage rates')
+
+    yield
+
```
Comment (quarckster) [inline cfme/tests/ssui/test_ssui_dashboard.py]: **REQUIRED**
Please remove these extra blank lines.

File: cfme/tests/ssui/test_ssui_dashboard.py [hunk 5]
```diff
@@ -27,6 +32,108 @@
...
+               record.cpu_usage_rate_average is not None or
+               record.derived_memory_used is not None or
+               record.net_usage_rate_average is not None or
+               record.disk_usage_rate_average):
+                return True
+
+        return False
+
+    wait_for(verify_records_rollups_table, [appliance, provider], timeout=3600,
+        fail_condition=False, message='Waiting for hourly rollups')
```
Comment (quarckster) [inline cfme/tests/ssui/test_ssui_dashboard.py]: **REQUIRED**
`fail_condition=False` is not needed

Comment (nachandr) [reply cfme/tests/ssui/test_ssui_dashboard.py]: @quarckster , I've addressed all your comments.

File: cfme/tests/ssui/test_ssui_dashboard.py [hunk 6]
```diff
@@ -1,14 +1,19 @@
 # -*- coding: utf-8 -*-
-
+import fauxfactory
```
Comment (mshriver) [inline cfme/tests/ssui/test_ssui_dashboard.py]: move this down to 2nd group, fauxfactory is pip installable

File: cfme/tests/ssui/test_ssui_dashboard.py [hunk 7]
```diff
@@ -1,14 +1,19 @@
...
 import pytest
+from datetime import date
 
+import cfme.intelligence.chargeback.assignments as cb
```
Comment (mshriver) [inline cfme/tests/ssui/test_ssui_dashboard.py]: **Optional**

Import assignments as something more descriptive.

**Comment by mshriver** [comment]
Discussed with psav, merging.

**Pull request merged**